agent/agent/src/main/java/com/microsoft/applicationinsights/agent/Agent.java

```diff
@@ -34,19 +35,33 @@ public class Agent {
   // this is to prevent the agent from loading and instrumenting everything twice
   // (leading to unpredictable results) when -javaagent:applicationinsights-agent.jar
   // appears multiple times on the command line
-  private static volatile boolean alreadyLoaded;
+  private static final AtomicBoolean alreadyLoaded = new AtomicBoolean(false);
 
   public static void premain(String agentArgs, Instrumentation inst) {
-    if (alreadyLoaded) {
+    if (alreadyLoaded.compareAndSet(true, true)) {
       return;
     }
 
-    if (Boolean.getBoolean("applicationinsights.debug.startupProfiling")) {
-      StartupProfiler.start();
+    try {
+      if (Boolean.getBoolean("applicationinsights.debug.startupProfiling")) {
+        StartupProfiler.start();
+      }
+
+      OpenTelemetryAgent.premain(agentArgs, inst);
+    } finally {
+      if (!agentIsInstalled()) {
+        alreadyLoaded.set(false);
+      }
```

Comment:
do you think this is needed? my thinking is that if it failed to install, may as well not try again

Reply:
Removed, as discussed